lib/buffer.js

```diff
@@ -637,6 +637,28 @@ Buffer.prototype.includes = function includes(val, byteOffset, encoding) {
   return this.indexOf(val, byteOffset, encoding) !== -1;
 };
 
+// Limit currently not implemented; reserved
+Buffer.prototype.split = function split(sep, limit) {
+  if (!limit)
+    limit = null;
+  if (!sep)
+    return [this];
+  if (!Buffer.isBuffer(sep))
+    throw new TypeError('First argument must be a buffer.');
+
+  return binding.split(this, sep, limit);
+};
+
+// not overriding prototype here, because of Uint8Array, REVIEW
+Buffer.join = function join(array, insert) {
+  if (!insert)
+    insert = null;
+  if (!Array.isArray(array) || !Buffer.isBuffer(array[0]))
```

Comment:
Do both checks in C++ and throw from there. No need to check `insert` here. Check using `HasInstance()` in C++ and throw from there.